### ISSUE TITLE
Yarpmanager buttons disabled when changing application tab

### ISSUE THREAD

**Issue by marco-monforte** [robotology/yarp#2355]
**Describe the bug**
Once you change the application tab in the yarpmanager, the buttons Play/Stop/Kill/Connect/Disconnect (both single module and for all modules) are disabled forever.

**To Reproduce**
1) Load an application from Entities -> The buttons are ok
2) Load a second application -> The buttons are still ok
3) Move back to the first application -> Buttons are disabled

**Expected behavior**

**Screenshots**

- When loading the first application:

![1](https://user-images.githubusercontent.com/32684998/92735669-e2ce8880-f379-11ea-9957-555208fbff69.png)

- After loading a second application and moving back to the first tab:

![2](https://user-images.githubusercontent.com/32684998/92735779-faa60c80-f379-11ea-9448-db830c340353.png)

**Configuration (please complete the following information):**
 - OS: Ubuntu 18.04.5 LTS
 - yarp version: YARP version 3.4.100+20200910.2+git1f9456379
-  compiler: gcc version 7.5.0

**Comment by pattacini** [comment]
I can confirm I've seen the same problem.

**Comment by Nicogene** [comment]
This issue is under investigation and the function causing this bug is the following:

https://github.com/robotology/yarp/blob/8ea25d2647b649b290be83a54ce0f2d094b59269/src/yarpmanager/src-manager/mainwindow.cpp#L909

In particular, there is this piece of code https://github.com/robotology/yarp/blob/8ea25d2647b649b290be83a54ce0f2d094b59269/src/yarpmanager/src-manager/mainwindow.cpp#L958-L969

Removing this should fix the issue but first I have to understand what was the rationale behind it before trimming it.

**Comment by Nicogene** [comment]
The `*_all` actions seem  to be handled only inside https://github.com/robotology/yarp/blob/8ea25d2647b649b290be83a54ce0f2d094b59269/src/yarpmanager/src-manager/mainwindow.cpp (as expected)

And here is the history of that file https://github.com/robotology/yarp/commits/8ea25d2647b649b290be83a54ce0f2d094b59269/src/yarpmanager/src-manager/mainwindow.cpp

The only recent commit is https://github.com/robotology/yarp/commit/41473510bc42960def8ffab8a8cdee6f01a558ad#diff-08d198d7b174504e27ad5e7f2cb2d19c that to me doesn't seem to interact with the buttons in the sidebar

**Comment by Nicogene** [comment]
It seems to me that there is another hidden issue, the refresh button is enabled when opening a new app(see https://github.com/robotology/yarp/blob/8ea25d2647b649b290be83a54ce0f2d094b59269/src/yarpmanager/src-manager/mainwindow.cpp#L644) But then is disabled when changing tab if the tab opened **IS NOT** a `Resource`(then if you switch between applications it get disabled)

https://github.com/robotology/yarp/blob/8ea25d2647b649b290be83a54ce0f2d094b59269/src/yarpmanager/src-manager/mainwindow.cpp#L951-L957

I have no idea what is the rationale behind this

**Comment by pattacini** [comment]
@Nicogene I would verify that this behavior is actually as you've read from the code.
Then, I would amend the code to make it behave as a traditional UX should prescribe.

**Comment by Nicogene** [comment]
> @Nicogene I would verify that this behavior is actually as you've read from the code.
Then, I would amend the code to make it behave as a traditional UX should prescribe.

Yes I first noticed it testing the manager

I checked out to `v3.2.0` and those bugs are not present, I will try with `git bisect`

**Comment by Nicogene** [comment]
Bisecting I found out that here is the faulty commit:

```
41473510bc42960def8ffab8a8cdee6f01a558ad is the first bad commit
commit 41473510bc42960def8ffab8a8cdee6f01a558ad
Author: Nicolo <nicolo.genesio@iit.it>
Date:   Fri Feb 28 15:30:43 2020 +0100

    yarpmanager: add builder tab.

 .../src-manager/applicationviewwidget.cpp          |  43 -------
 .../src-manager/applicationviewwidget.h            |   5 -
 .../src-manager/applicationviewwidget.ui           | 139 +++++++++++----------
 src/yarpmanager/src-manager/mainwindow.cpp         |  97 --------------
 src/yarpmanager/src-manager/mainwindow.h           |   2 -
 5 files changed, 74 insertions(+), 212 deletions(-)
```

In particular I trimmed away an `else` statement. Now the PR contains the right fix

**Comment by drdanz** [comment]
Fixed by @Nicogene in #2367
